### PR TITLE
Support adding and removing entities to devices

### PR DESCRIPTION
**Pull request by postlund**
This is not what I should be working on, but I had most of the code sitting on a branch so I decided to finish it.

New  entities can be added via + button under Integrations (like when adding a new device, but select an existing device instead) and entities can be removed from Options.

### CODE REVIEW

**Comment by ultratoto14** [comment]
Hi @postlund, just tested it, here are my remarks:

- When adding a new entity to an existing one, the already used DP ids are proposed.
- At the end the label is not what is expected, "Abandonné" means "canceled"
Canceled
- At the same time i have an exception in the logs, but the entity was well created:
```
2020-11-26 10:53:01 ERROR (MainThread) [homeassistant.config_entries] Error unloading entry Tempo for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 321, in async_unload
    result = await component.async_unload_entry(hass, self)  # type: ignore
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 72, in async_unload_entry
    return await hass.data[DOMAIN].async_unload_entry(entry)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 172, in async_unload_entry
    raise ValueError("Config entry was never loaded!")
ValueError: Config entry was never loaded!
```
- When trying to add a new entity, yaml defined entries are proposed, if i choose one i have an error, nothing in the logs
Error

**Comment by postlund** [comment]
Thanks for testing @ultratoto14! Let's see if I can answer...

Hmm, that's strange. Is it always like that or only when you add a new entity to a device that you added during the same "session"? Because I know that the browser sometimes cache values and fill those in for some reason. I at least don't have any code for auto-filling values (that's part of #125, which isn't merged yet).

That's a limitation of the entry flow framework. There's only two ways out: creating a new config entry or aborting (yielding the "canceled" message). Since we already have a config entry, that leaves out the first one (which also would request the user to assign the device to a room again), leaving only the abort method. Fact is that this is the de-facto way of doing things like this, e.g. when reconfiguring credentials for a device. So you will see it in the built in components as well.

The exception I don't know the reason for, will have to look into that.

Right, good catch. I will make sure to filter out devices set up via YAML.

**Comment by ultratoto14** [comment]
@postlund should I retest this one ?

**Comment by postlund** [comment]
@ultratoto14 Yes please, forgot to ping you about it.

**Review by ultratoto14 — APPROVED** [review]
Code looks good 👍 
Tests are Ok:

- Adding new device 👍  
- Adding entity to device 👍 
- Removing entity from device 👍 
- YAML devices not listed 👍 
- Already configured and not yet configured are there in list 👍

**Comment by postlund** [comment]
Great! 👍 Let's merge before I charge my mind.

File: custom_components/localtuya/__init__.py [hunk 1]
```diff
@@ -142,7 +136,7 @@ async def _handle_reload(service):
 
     async def _handle_set_dp(event):
         """Handle set_dp service call."""
-        entry = _entry_by_device_id(event.data[CONF_DEVICE_ID])
+        entry = async_config_entry_by_device_id(event.data[CONF_DEVICE_ID])
```
Comment (ultratoto14) [inline custom_components/localtuya/__init__.py:139]: Seems the `hass` is missing.

**Review by ultratoto14 — CHANGES_REQUESTED** [review]

File: custom_components/localtuya/__init__.py (hunk 1 above)
Comment (postlund) [reply custom_components/localtuya/__init__.py:139]: Ah, right!

**Review by ultratoto14 — APPROVED** [review]

**Comment by postlund** [comment]
🎉

**Pull request merged**